File: app/src/main/java/cz/johnyapps/cheers/adapters/BeverageFragmentAdapter.java
package cz.johnyapps.cheers.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import cz.johnyapps.cheers.entities.BeverageCategory;
import cz.johnyapps.cheers.fragments.BeverageCategoryFragment;

public class BeverageFragmentAdapter extends FragmentStateAdapter {
    @NonNull
    private final BeverageCategory[] beverageCategories;

    public BeverageFragmentAdapter(@NonNull Fragment fragment,
                                   @NonNull BeverageCategory[] beverageCategories) {
        super(fragment);
        this.beverageCategories = beverageCategories;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return new BeverageCategoryFragment(beverageCategories[position]);
    }

    @Override
    public int getItemCount() {
        return beverageCategories.length;
    }
}

File: app/src/main/java/cz/johnyapps/cheers/adapters/BeverageFragmentAdapter.kt
package cz.johnyapps.cheers.adapters

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import cz.johnyapps.cheers.entities.BeverageCategory
import cz.johnyapps.cheers.fragments.BeverageCategoryFragment
import cz.johnyapps.cheers.viewmodels.BeverageCategoriesViewModel

class BeverageFragmentAdapter(fragment: Fragment, private val viewModel: BeverageCategoriesViewModel): FragmentStateAdapter(fragment) {
    private var fragments = ArrayList<Fragment>()

    override fun getItemCount(): Int {
        return BeverageCategory.values().size
    }

    override fun createFragment(position: Int): Fragment {
        val fragment = BeverageCategoryFragment(viewModel, BeverageCategory.values()[position])
        fragments.add(fragment)
        return fragment
    }

    fun getFragment(position: Int): Fragment? {
        if (position in 0 until itemCount) {
            return fragments[position]
        }

        return null
    }
}